firefox-ios/Client/TabManagement/Legacy/LegacyTabManager.swift

```diff
@@ -662,7 +662,7 @@ class LegacyTabManager: NSObject, FeatureFlaggable, TabManager, TabEventHandler
         guard !backupCloseTabs.isEmpty else { return }
         tabs = backupCloseTabs
         storeChanges()
-        backupCloseTabs = [Tab]()
+        backupCloseTabs.removeAll(keepingCapacity: true)
```

Comment:
What is the intention of this change?

Reply:
AFAIU: the previous code deallocates the old collection and creates a new instance which then has to be resized in memory as new items are added. This code just clears it and preserves the memory already allocated which is sometimes more performant, depending on how it's used. But because you asked about it I looked closer and I'm not sure it matters much because we don't actually insert the tabs into the collection, when `backupCloseTabs` is updated we just assign the `tabs` to `backupTabs` directly (which, because Array is copy-on-write, means that `backupTabs` is just going to point to the existing `tabs` collection until it's modified). Since it's just a collection of pointers to class instances in memory elsewhere I doubt it makes much difference anyway, even for 100's of tabs. I'll revert this change since it's not really necessary.